pkgs/development/libraries/libtiff/default.nix

```diff
@@ -63,6 +64,7 @@ stdenv.mkDerivation rec {
 
   # TODO: opengl support (bogus configure detection)
   propagatedBuildInputs = [
+    lerc
```

Comment:
Why does it need to be propagated?

Reply:
It wasn't clear to us as well.

From [GDAL documentation](https://gdal.org/drivers/raster/gtiff.html#creation-options)

"""
LERC and LERC_DEFLATE are available only when using internal libtiff for GDAL < 3.3.0. Since GDAL 3.3.0, LERC compression is also available **when building GDAL against external libtiff** >= 4.3.0, **built itself against https://github.com/esri/lerc**
"""

Reply:
I'm not sure why the current list is propagated, but we shouldn't add new propagated inputs without knowing why.

Reply:
True, I understand what you are saying now.

Reply:
With 

```
diff --git a/pkgs/development/libraries/libtiff/default.nix b/pkgs/development/libraries/libtiff/default.nix
index 1614748afcb1..e087e861a06b 100644
--- a/pkgs/development/libraries/libtiff/default.nix
+++ b/pkgs/development/libraries/libtiff/default.nix
@@ -63,7 +63,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoreconfHook pkg-config sphinx ];

   # TODO: opengl support (bogus configure detection)
-  propagatedBuildInputs = [
+  buildInputs = [
     lerc
     libdeflate
     libjpeg
```

build of `libgeotiff` fails with

```
configure: error: cannot find zlib.h, bailing out
```

Reply:
If I add `zlib` in to libgeotiff `nativeBuildInputs`

```
diff --git a/pkgs/development/libraries/libgeotiff/default.nix b/pkgs/development/libraries/libgeotiff/default.nix
index 9b160e6b29e8..bf2d6a42dbc3 100644
--- a/pkgs/development/libraries/libgeotiff/default.nix
+++ b/pkgs/development/libraries/libgeotiff/default.nix
@@ -32,6 +32,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     autoreconfHook
     pkg-config
+    zlib
   ];

   buildInputs = [ libtiff proj ];
```

build of `libgeotiff` will finish successfully.

@alyssais  Looks like confirmation of negative impact of `propagatedBuildInputs` in `libtiff` which we should fix.

Reply:
Actually, `zlib` should go to `buildInputs` of `libgeotiff`.

Reply:
Draft PR to fix propagation of libtiff build inputs created https://github.com/NixOS/nixpkgs/pull/291657

Reply:
I've changed `lerc` to a `buildInput`. Does it make sense to merge this before #292568, or should they go in together?